tests/test_builder.py

```diff
@@ -0,0 +1,111 @@
+import sys
+from mlir import parse_string
+from mlir.builder import IRBuilder
+from mlir.builder import Reads, Writes, Isa
+from mlir.dialects.affine import AffineLoadOp
+from mlir.dialects.standard import AddfOperation
+
+
+def test_saxpy_builder():
+    builder = IRBuilder()
+    F64 = builder.F64
+    Mref1D = builder.MemRefType(shape=(None, ), dtype=F64)
+
+    mlirfile = builder.make_mlir_file()
+    module = mlirfile.module
+
+    with builder.goto_block(builder.make_block(module.region)):
+        saxpy_fn = builder.function("saxpy")
+
+    block = builder.make_block(saxpy_fn.region)
+    builder.position_at_entry(block)
+
+    a, x, y = builder.add_function_args(saxpy_fn, [F64, Mref1D, Mref1D])
+    c0 = builder.index_constant(0)
+    n = builder.dim(x, c0, builder.INDEX)
+
+    f = builder.affine_for(0, n)
```

Comment:
General question about dialects and the IR builder: shouldn't dialects somehow extend the general builder? right now it looks like new dialects would all add to the `IRBuilder` class.

Reply:
My aim was to provide a builder for the core-dialects. I could have had a builder for each of the core dialect, but I figured it isn't buying us anything.

A user could just derive this and add ops correpsonding to their custom dialect.

Reply:
shouldn't the default dialects serve as an example on how to extend then, i.e., with `builder.affine.for`? As opposed to numba or other IR builders, MLIR revolves around its extensible dialects, so the API should somewhat match. I'd propose one can construct an IR builder as follows:
```python
builder = IRBuilder(affine=AffineIRBuilder(), ...) # for other dialects
```
then, the individual dialects can be used as `builder.affine.load(...)` or `builder.linalg.dot`.

Reply:
I quite liked this idea of namespacing the dialect ops. Implemented, thanks a lot!